device/device.go

```diff
@@ -0,0 +1,99 @@
+package device
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/josephroberts/edge-node-manager/radio"
+)
+
+type SupportedDevice string
+
+const (
+	NRF51822 SupportedDevice = "NRF51822"
+	ESP8266  SupportedDevice = "ESP8266"
+	MICROBIT SupportedDevice = "MicroBit"
+)
+
+func Create(d SupportedDevice) DeviceInterface {
+	switch d {
```

Comment:
Yeah I agree, still need to figure out a good pattern that allows a user to simply add a new device type (and possibly a radio if needed)